File: JS/Programmers/Level1/12 예산.js
function solution(d, budget) {
  var answer = 0;

  var allotment = 0;
  d.sort((a, b) => a - b);

  for (var i = 0; i < d.length; i++) {
    if (allotment + d[i] <= budget) {
      allotment += d[i];
      answer++;
      d[i] = 0;
    }
  }

  return answer;
}

File: C++/Programmers/Level1/12 예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> d, int budget) {
    // d = 신청한 금액
    // budget = 예산

    int answer = 0;
    int allotment = 0;
    sort(d.begin(), d.end());

    for(int i=0; i<d.size(); i++){
        if(allotment + d[i] <= budget){
            allotment += d[i];
            answer++;
            d[i] = 0;
        }
    }
    
    return answer;
}